rct229/rules/section6/section6rule4.py

```diff
@@ -0,0 +1,134 @@
+                    building_segment_p
+                )
+            }
+
+        class ZoneRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(
+                    Section6Rule7.BuildingRule.BuildingSegmentRule.ZoneRule,
+                    self,
+                ).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    each_rule=Section6Rule7.BuildingRule.BuildingSegmentRule.ZoneRule.SpaceRule(),
+                    index_rmr="baseline",
+                    list_path="spaces[*]",
+                )
+
+            def create_data(self, context, data=None):
+                zone_b = context.baseline
+
+                # We will need this after Weili's update to table_G3_7_lookup()
+                return {"avg_zone_ht_b": get_avg_zone_ht(zone_b)}
+
+        class SpaceRule(RuleDefinitionBase):
+            def __init__(self):
+                super(
+                    Section6Rule7.BuildingRule.BuildingSegmentRule.SpaceRule,
+                    self,
+                ).__init__(
+                    fail_msg="P_RMR lighting status type is as-designed or as-existing. But lighting space type in B_RMR is not specified.",
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                )
+
+            def get_calc_vals(self, context, data=None):
+                space_b = context.baseline
+                space_p = context.proposed
+                total_space_lpd_b = sum(
+                    find_all("interior_lighting[*].power_per_area", space_b)
+                )
+                space_lighting_status_type_p = data[
+                    "building_segment_lighting_status_type_dict_p"
+                ][space_p["id"]]
+                lpd_allowance_b = (
+                    table_G3_7_lookup(
+                        space_b["lighting_space_type"], data["avg_zone_ht_b"]
+                    )
+                    if "lighting_space_type" in space_b
+                    else table_G3_7_lookup(OFFICE_OPEN_PLAN, data["avg_zone_ht_b"])
+                )
+
+                return {
+                    "total_space_lpd_b": total_space_lpd_b,
+                    "space_lighting_status_type_p": space_lighting_status_type_p,
+                    "lpd_allowance_b": lpd_allowance_b,
+                }
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                space_b = context.baseline
+                lighting_space_type_b = space_b.get("lighting_space_type")
+
+                space_lighting_status_type_p = calc_vals["space_lighting_status_type_p"]
+                total_space_lpd_b = calc_vals["total_space_lpd_b"]
+                lpd_allowance_b = calc_vals["lpd_allowance_b"]
+
+                return (
+                    # Not Case 1
+                    not (
+                        space_lighting_status_type_p
+                        == LightingStatusType.AS_DESIGNED_OR_AS_EXISTING
```

Comment:
I belive ``` or space_lighting_status_type_p == LightingStatusType.NOT_YET_DESIGNED_OR_MATCH_TABLE_9_5_1``` should be added to take the Case 4 into account. Could you check this out?